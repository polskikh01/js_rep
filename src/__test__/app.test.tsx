import React from 'react';
import { mount } from "enzyme";
import { describe, it, expect } from '@jest/globals'

import App from '../app';

describe('Тестирование всего приложения', () => {
    it('Тестируем рендер App', () => {
        expect.assertions(1)
        const component = mount(<App/>)

        expect(component).toMatchSnapshot()
    });
})