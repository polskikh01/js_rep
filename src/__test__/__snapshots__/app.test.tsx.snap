// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование всего приложения Тестируем рендер App 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter
      basename="/js_rep"
    >
      <Router
        history={
          Object {
            "action": "REPLACE",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "s46he2",
              "pathname": "/main",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Dashboadr>
          <Switch>
            <Route
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/main",
                  "url": "/main",
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "s46he2",
                  "pathname": "/main",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/main"
            >
              <Main>
                <div
                  id="wrapper"
                >
                  <div
                    className="focusWrapper"
                    id="focus-wrapper"
                  >
                    <div
                      className="page"
                    >
                      <Head>
                        <header
                          className="mHeader"
                        >
                          <div
                            className="menuHeader"
                          >
                            <div
                              className="menu"
                            >
                              <Link
                                className="logo"
                                to="/main"
                              >
                                <LinkAnchor
                                  className="logo"
                                  href="/js_rep/main"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="logo"
                                    href="/js_rep/main"
                                    onClick={[Function]}
                                  >
                                    <span
                                      className="lSText"
                                    >
                                      Calabash
                                    </span>
                                  </a>
                                </LinkAnchor>
                              </Link>
                              <nav
                                className="navP"
                              >
                                <Nav
                                  link="/product"
                                >
                                  <Link
                                    className="navButt"
                                    to="/product"
                                  >
                                    <LinkAnchor
                                      className="navButt"
                                      href="/js_rep/product"
                                      navigate={[Function]}
                                    >
                                      <a
                                        className="navButt"
                                        href="/js_rep/product"
                                        onClick={[Function]}
                                      >
                                        <span />
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                </Nav>
                                <Nav
                                  link="/shippingAndPayment"
                                >
                                  <Link
                                    className="navButt"
                                    to="/shippingAndPayment"
                                  >
                                    <LinkAnchor
                                      className="navButt"
                                      href="/js_rep/shippingAndPayment"
                                      navigate={[Function]}
                                    >
                                      <a
                                        className="navButt"
                                        href="/js_rep/shippingAndPayment"
                                        onClick={[Function]}
                                      >
                                        <span />
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                </Nav>
                                <Nav
                                  link="/contacts"
                                >
                                  <Link
                                    className="navButt"
                                    to="/contacts"
                                  >
                                    <LinkAnchor
                                      className="navButt"
                                      href="/js_rep/contacts"
                                      navigate={[Function]}
                                    >
                                      <a
                                        className="navButt"
                                        href="/js_rep/contacts"
                                        onClick={[Function]}
                                      >
                                        <span />
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                </Nav>
                              </nav>
                              <div
                                className="basket"
                              >
                                <Link
                                  to="/basket"
                                >
                                  <LinkAnchor
                                    href="/js_rep/basket"
                                    navigate={[Function]}
                                  >
                                    <a
                                      href="/js_rep/basket"
                                      onClick={[Function]}
                                    >
                                      <img
                                        alt="Basket"
                                        src="FILE_MOCK"
                                      />
                                    </a>
                                  </LinkAnchor>
                                </Link>
                              </div>
                            </div>
                          </div>
                        </header>
                      </Head>
                      <div
                        className="wrap"
                      >
                        <div
                          className="line"
                        >
                          <section>
                            <div
                              className="info"
                            >
                              <div
                                className="infoPhoto"
                              />
                              <div
                                className="infoText"
                              >
                                <div
                                  className="text"
                                >
                                  <div
                                    className="textT"
                                  >
                                    <span
                                      className="lSText"
                                    >
                                      Calabash
                                    </span>
                                  </div>
                                  <div
                                    className="textT"
                                  >
                                    <span
                                      className="lSTextB"
                                    >
                                       
                                      <br />
                                       
                                    </span>
                                  </div>
                                  <div
                                    className="textT"
                                  >
                                    <Link
                                      to="/product"
                                    >
                                      <LinkAnchor
                                        href="/js_rep/product"
                                        navigate={[Function]}
                                      >
                                        <a
                                          href="/js_rep/product"
                                          onClick={[Function]}
                                        >
                                          <span
                                            className="lSTextA"
                                          />
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </section>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </Main>
            </Route>
          </Switch>
        </Dashboadr>
      </Router>
    </BrowserRouter>
  </Provider>
</App>
`;
